scripts/restore

```diff
@@ -1,51 +1,94 @@
+# 	# Get the _common.sh file if it's not in the current directory
+# 	cp ../settings/scripts/_common.sh ./_common.sh
+# 	chmod a+rx _common.sh
+# fi
+# source _common.sh
 source /usr/share/yunohost/helpers
 
-# Récupère les infos de l'application.
+#=================================================
+# LOAD SETTINGS
+#=================================================
+
 app=$YNH_APP_INSTANCE_NAME
 
 domain=$(ynh_app_setting_get $app domain)
-path=$(ynh_app_setting_get $app path)
+path_url=$(ynh_app_setting_get $app path)
 admin=$(ynh_app_setting_get $app admin)
 final_path=$(ynh_app_setting_get $app final_path)
-db_user=$(ynh_app_setting_get $app db_user)
-db_pwd=$(ynh_app_setting_get $app mysqlpwd)
+db_name=$(ynh_app_setting_get $app db_name)
+
+#=================================================
+# CHECK IF THE APP CAN BE RESTORED
+#=================================================
 
-if [ -d $final_path ]; then
-    echo "There is already a directory: $final_path " >&2
-    ynh_die
-fi
+ynh_webpath_available $domain $path_url \
+	|| ynh_die "Path not available: ${domain}${path_url}"
+test ! -d $final_path \
+	|| ynh_die "There is already a directory: $final_path "
 
-# Restore Nginx
-conf=/etc/nginx/conf.d/$domain.d/$app.conf
-if [ -f $conf ]; then
-    echo "There is already a nginx conf file at this path: $conf " >&2
-    ynh_die
-fi
-sudo cp -a ./nginx.conf $conf
+#=================================================
+# STANDARD RESTORATION STEPS
+#=================================================
+# RESTORE THE NGINX CONFIGURATION
+#=================================================
+
+ynh_restore_file "/etc/nginx/conf.d/$domain.d/$app.conf"
+
+#=================================================
+# RESTORE THE APP MAIN DIR
+#=================================================
+
+ynh_restore_file "$final_path"
+
+#=================================================
+# RESTORE THE MYSQL DATABASE
+#=================================================
+
+db_pwd=$(ynh_app_setting_get $app mysqlpwd)
+ynh_mysql_setup_db $db_name $db_name $db_pwd
+ynh_mysql_connect_as $db_name $db_pwd $db_name < ./db.sql
 
-# Restore YunoHost parameters
-sudo cp -a ./yunohost/. /etc/yunohost/apps/$app/
+#=================================================
+# RECREATE THE DEDICATED USER
+#=================================================
 
-# Restore sources & data
-sudo mkdir -p $final_path
-sudo cp -a ./sources/* $final_path/
+# Create the dedicated user (if not existing)
+ynh_system_user_create $app
 
-ynh_mysql_create_db $db_user $db_user $db_pwd
-mysql --debug-check -u $db_user -p$db_pwd $db_user < ./backupdb.sql
+#=================================================
+# RESTORE USER RIGHTS
+#=================================================
 
-# Copy dedicated php-fpm process from backup folder to the right location
-sudo cp -a ./php-fpm.conf /etc/php5/fpm/pool.d/$app.conf
+# config.inc.php contains sensitive data, restrict its access
+chown root:$app $final_path/config.inc.php
```

Comment:
How does restoration work? Does it attribute the same file rights as backed-up files?
So wouldn't that file be already owned by root?

Reply:
The restoration should keep the root user and its uid, but not the $app user because it's created just before at the line 72.

Reply:
Indeed! That's maybe an idea for the core: why not systematically create/delete a user named after the app instance name?